### PR TITLE
[ENG-533] Update navigation

### PR DESCRIPTION
**Pull request by niikeec**

### CODE REVIEW

File: interface/app/$libraryId/Layout/Sidebar/Link.tsx [hunk 1]
```diff
@@ -20,20 +20,25 @@ const styles = cva(
...
+	...props
+}: PropsWithChildren<NavLinkProps & { disabled?: boolean }>) => {
 	const os = useOperatingSystem();
+	const location = useLocation();
 
 	return (
 		<NavLink
-			{...props}
 			onClick={(e) => (props.disabled ? e.preventDefault() : props.onClick?.(e))}
+			replace={!!location.pathname.match(/^\/[\w\d-]+\/settings\//)}
```
Comment (Brendonovich) [inline interface/app/$libraryId/Layout/Sidebar/Link.tsx:33]: Leveraging `useMatch` here would probably be nicer

**Review by Brendonovich — DISMISSED** [review]
This whole system seems rather complex when we already have a router - could a combination of `replace: true` and checking `history` do the job?

File: interface/app/$libraryId/TopBar/SearchBar.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ export default () => {
 	}, 300);
 
 	useEffect(() => {
-		updateParams(value);
+		if (searchPath.pathname === location.pathname) updateParams(value);
 	}, [value]);
```
Comment (utkubakir) [inline interface/app/$libraryId/TopBar/SearchBar.tsx:43]: I think you need to add `searchPath` & `location`

File: interface/app/onboarding/Layout.tsx [hunk 3]
```diff
@@ -19,20 +25,26 @@ export const Component = () => {
 	const debugState = useDebugState();
 	const navigate = useNavigate();
 
+	const libraries = useCachedLibraries();
+	const library =
```
Comment (utkubakir) [inline interface/app/onboarding/Layout.tsx:29]: Don't we have a way to get the currentLibrary? defaulting the first one seems weird

Comment (niikeec) [reply interface/app/onboarding/Layout.tsx:29]: Only defaults to first if it can't find the current one.

**Review by utkubakir — APPROVED** [review]

**Pull request merged**